src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -4067,21 +4052,22 @@ tapOffsetEvery offset n (Fold fstep initial extract) (Stream step state) =
     step' gst (TapOffTapping acc st count) = do
         r <- step gst st
         case r of
-            Yield x s ->
-                if count <= 0
-                -- XXX Abstract the then branch?
-                then do
-                    acc1 <- fstep acc x
-                    return
-                        $ case acc1 of
-                              FL.Partial sres ->
-                                  Yield x $ TapOffTapping sres s (n - 1)
-                              FL.Done _ -> Yield x (TapOffDone s)
-                else return $ Yield x $ TapOffTapping acc s (count - 1)
+            Yield x s -> do
```

Comment:
I don't see any advantage in this refactor.

Reply:
Just floats out the Yield, has lesser yield points, if that helps the compiler.